### DISCUSSION TITLE
Send File to client with Adonis

### DISCUSSION THREAD

**Discussion by Hakamate** [adonisjs/core#1344] [Help] [answered]
Hello everyone,

I try to send back a file to a nuxtjs client. But i don't really know how can i do that.

Can you bring me links ? advices ? Or maybe code example ?

**Comment by thetutlage** [comment, marked as answer]
Your question is little vague. What you mean by sending file to Nuxt? Do you want the user of your app to download a file?

**Reply by Hakamate** [reply]
Yes i want the user download a file from Adonis

**Reply by thetutlage** [reply]
You have to redirect them to the AdonisJS route and use the `response.download()` method to send the file as response.

Do note: Ajax requests don't allow downloading files, you have to redirect the user to a url handled by AdonisJS

**Reply by mortenko** [reply]
probably you need to do some work also on frontend : https://stackoverflow.com/questions/3749231/download-file-using-javascript-jquery.  (Third post)

**Reply by Hakamate** [reply]
@mortenko actually here is the response from the server : 

![response](https://user-images.githubusercontent.com/33601556/88548518-7ee93e00-d01f-11ea-90b2-fe32f31f97af.png)

There is a way to translate it into a downloadable file ?

**Reply by thetutlage** [reply]
Already told you, you cannot stream files in an Ajax request. Please read the above comment

**Reply by Hakamate** [reply]
Ok, I got it, that because Axios can't consume download stream. Thank you @thetutlage i will use fetch or a library to allow it ;)

**Comment by Hakamate** [comment]
For the context, i'm actually trying to make a youtube downloader

Nuxt Code (client) :
```
sendURL: async function(url){
      console.log(url)
      await axios.post(`http://127.0.0.1:3333/api/download`, {params:{URL:url}})
        .catch(error => {
          console.log(error);
        })
        .then(response => console.log(response))
    }
```

Adonis Code :

```

const fs = require('fs');
const Helpers = use('Helpers');
const youtubedl = require('youtube-dl');

class DownloadController {
download({ request, response }) {
        // response.setHeader('Content-Type', 'text/event-stream');
        console.log("cc")
        response.implicitEnd = false

        let title = null
        const options = []
        const req = request.all()
        var link = req.params.URL

        const video = youtubedl(link, ["--format=bestaudio"], {cwd: __dirname});
       
        youtubedl.getInfo(link, options, function(err, info){
            if(err) throw err
            title = info.title
            const target = Helpers.publicPath(title + ".mp3")
    
            video.pipe(
                fs.createWriteStream(target)
            )

            video.on('end', function() {
                console.log(title, ' finished downloading !')
                // response.send(target)
                // fs.readFile(target, (error, contents) => {
                //     response.download(contents)
                //   })
                response.download(target)
            })
        })
    }
}
```
